lib/board.rb

```diff
@@ -0,0 +1,104 @@
+require 'pry'
+require_relative "player.rb"
+
+class Board
+	attr_accessor :cells
+
+	def initialize
+		@cells = []
+		reset!
+		#binding.pry
+	end
+
+	 #combinations where the other player is 1 turn from winning
+  BLOCK_COMBO = [
+    {[0,1] => 2},
+    {[3,4] => 5},
+    {[6,7] => 8},
+    {[1,2] => 0},
+    {[4,5] => 3},
+    {[7,8] => 6},
+    {[0,3] => 6},
+    {[2,5] => 8},
+    {[1,4] => 7},
+    {[0,4] => 8},
+    {[2,4] => 6},
+    {[4,8] => 0},
+    {[4,6] => 2},
+    {[5,8] => 2},
+    {[4,7] => 1},
+    {[3,6] => 0},
+  ]
+
+  def block?(token)
+  	a = nil
+  	BLOCK_COMBO.each do |hash|
+      hash.each do |combo, mov|
+        if position(combo[0]+1) != token && position(combo[0]+1) != " " && position(combo[1]+1) != token && position(combo[1]+1) != " " && !taken?(mov+1)
+        	binding.pry
+        	a=mov
+        end
+      end
+    end
+    a
+  end
+
+	def reset!
+		@cells = []
+		9.times do 
+			@cells << " "
+		end
```

Comment:
Another way to instantiate an Array with `i` values is `Array.new(9, " ")` (see [Array.new](http://ruby-doc.org/core-2.2.0/Array.html#method-c-new)), where `9` is the number of elements in the array and `" "` is the default value for the elements. It's more concise and will prevent errors from repeatedly typing `" "`.